lua/ttt2/libraries/migrations.lua

```diff
@@ -9,114 +9,65 @@ end
 
 migrations = {}
 migrations.folderPath = "terrortown/migrations/"
-migrations.baseFolderPath = migrations.folderPath .. "base/"
 migrations.databaseName = "ttt2_migrations"
 migrations.savingKeys = {
 	executedAt = {
 		typ = "string",
 		default = ""
-	},
-	statusMessage = {
-		typ = "string",
-		default = "Success"
-	},
+	}
 }
 
 sql.CreateSqlTable(migrations.databaseName, migrations.savingKeys)
 migrations.orm = orm.Make(migrations.databaseName)
 
-migrations.migrations = {}
-migrations.isLoaded = false
-
 ---
 -- Runs all missing forward gamemode migrations.
--- @return boolean `true` if the desired version was successfully migrated and `false` in case of an error.
 -- @realm shared
 function migrations.Apply()
-	if not migrations.isLoaded then
-		migrations.Load()
-	end
+	local files = file.Find(migrations.folderPath .. "*.lua", "LUA", "nameasc") or {}
 
-	local migrationSuccess = true
-	local errorMessage = ""
+	local fileExecutionCounter = 0
 
-	for fileName, migration in SortedPairs(migrations.migrations) do
-		if not IsValid(migration) then continue end
+	-- Add Hook to OnLuaError to catch if migrations fail
+	local isSuccess = true
+	local hookName = "OnLuaError"
+	local hookIdentifier = "TTT2MigrationErrorCatcher"
+	hook.Add(hookName, hookIdentifier, function(error, realm, stack, name, id)
+		isSuccess = false
+	end)
```

Comment:
Maybe use ProtectedCall here instead?
And use `ProtectedCall(include(...))` because we still want it to print errors etc but this might be much cleaner than relying on the hook

Reply:
I can test that, but pcall already didnt work, because the include call itself is without error.

Reply:
Does also not work.

Reply:
Mhh if include is really not catchable, then we might just consider changing the migrations to just contain an anonymous function which is returned at the end of the file.
Then we could pcall the function and it would be cleaner than using the hook.

Reply:
Done.